libs/hooks/open-telemetry/src/lib/open-telemetry-hook.ts

```diff
@@ -1,65 +1,37 @@
-import {
-  Hook,
-  HookContext,
-  EvaluationDetails,
-  FlagValue,
-} from '@openfeature/js-sdk';
-import { Span, Tracer, trace } from '@opentelemetry/api';
+import { Hook, HookContext, EvaluationDetails, FlagValue } from '@openfeature/js-sdk';
+import { trace } from '@opentelemetry/api';
 
-const SpanProperties = Object.freeze({
-  FLAG_KEY: 'feature_flag.flag_key',
+const eventName = 'feature_flag';
+const SpanEventProperties = Object.freeze({
+  FLAG_KEY: 'feature_flag.key',
   PROVIDER_NAME: 'feature_flag.provider_name',
-  VARIANT: 'feature_flag.evaluated_variant',
-  VALUE: 'feature_flag.evaluated_value',
+  VARIANT: 'feature_flag.variant',
 });
 
 export class OpenTelemetryHook implements Hook {
-  private spanMap = new WeakMap<HookContext, Span>();
-  private tracer: Tracer;
-
-  constructor() {
-    this.tracer = trace.getTracer(
-      '@openfeature/open-telemetry-hook',
-      '5.0.0' // x-release-please-version
-    );
-  }
-
-  before(hookContext: HookContext) {
-    const span = this.tracer.startSpan(
-      `${hookContext.providerMetadata.name} ${hookContext.flagKey}`,
-      {
-        attributes: {
-          [SpanProperties.FLAG_KEY]: hookContext.flagKey,
-          [SpanProperties.PROVIDER_NAME]: hookContext.providerMetadata.name,
-        },
+  after(hookContext: HookContext, evaluationDetails: EvaluationDetails<FlagValue>) {
+    const currentTrace = trace.getActiveSpan();
+    if (currentTrace) {
+      let variant = evaluationDetails.variant;
+
+      if (!variant) {
+        if (typeof evaluationDetails.value === 'string') {
+          variant = evaluationDetails.value;
+        } else {
+          variant = JSON.stringify(evaluationDetails.value);
```

Comment:
This will break with circular structures.

Reply:
Can the evaluation details value ever be a circular structure?

Reply:
It's JS so at the end of the day, providers can return whatever they want if they want to cast unsafely.

Reply:
If a circular reference is being provided, they can specify how it should be serialized with the `toJSON` method.

> If the value has a toJSON() method, it's responsible to define what data will be serialized. Instead of the object being serialized, the value returned by the toJSON() method when called will be serialized. JSON.stringify() calls toJSON with one parameter, the key

[Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)

If we are concerned about this not being straightforward, the actual interface could be updated to hint at the optional `toJSON` property.

Reply:
I'm not too concerned about it TBH, I just thought it was worth pointing out.

Reply:
Since this is directly on EvaluationDetails it only impacts providers.

I don't see this as likely but it's possible. Wouldn't hurt to be ahead of it.

Reply:
Thinking about it more, I think you should ignore this possibility.